src/util/postgres_client.js

```diff
@@ -707,6 +703,42 @@ class PostgresTable {
         return _do_query(client || this.get_pool(), q, 0);
     }
 
+   /**
+     * executeSQL takes a raw SQL query and params and runs it against
+     * the database. If `query_name` is passed then it prepares a
+     * statement on the first execution while the further executions
+     * will re-utilize the prepared statement (pre-parsed).
+     * 
+     * @template T
+     * 
+     * @param {string} query 
+     * @param {Array<any>} params 
+     * @param {{
+     *  query_name?: string,
+     * }} [options = {}]
+     * 
+     * @returns {Promise<import('pg').QueryResult<T>>}
+     */
+    async executeSQL(query, params, options = {}) {
+        /** @type {Pool} */
+        const pool = this.get_pool();
+        const client = await pool.connect();
+
+        const q = {
+            text: query,
+            values: params,
+        };
+
+        if (options.query_name) {
+            q.name = options.query_name;
+        }
+
+        const res = await _do_query(client, q, 0);
```

Comment:
why not to use `this.single_query` instead of `_do_query` directly?

Reply:
I will fix this, thank you.

Reply:
Ah right! The reason I didn't use that was because I wanted to expose prepared statements in the interface so that we can let postgres cache those statements which are executed too frequently (lifecycles ones aren't one of those).